src/util/config.js

```diff
@@ -54,12 +54,17 @@ const config: Config = {
         return /^((https?:)?\/\/)?api\.mapbox\.c(n|om)(\/mapbox-gl-js\/)(.*$)/i;
     },
     get EVENTS_URL() {
-        if (!this.API_URL) { return null; }
-        if (this.API_URL.indexOf('https://api.mapbox.cn') === 0) {
-            return 'https://events.mapbox.cn/events/v2';
-        } else if (this.API_URL.indexOf('https://api.mapbox.com') === 0) {
-            return 'https://events.mapbox.com/events/v2';
-        } else {
+        if (!config.API_URL) { return null; }
+        try {
+            const url = new URL(config.API_URL);
```

Comment:
Fix for CodeQL scanning alerts

* https://github.com/mapbox/mapbox-gl-js/security/code-scanning/19
* https://github.com/mapbox/mapbox-gl-js/security/code-scanning/20

Reply:
@stepankuzmin should we land this in a separate PR to unblock PR checks for other work?

Reply:
@mourner This is a separate issue. CodeQL is triggered on these lines after the code change. We should revert this PR to unblock the incoming PRs https://github.com/mapbox/mapbox-gl-js/pull/12562